ocaml/otherlibs/stdlib_alpha/capsule.mli

```diff
@@ -40,25 +40,68 @@
     this interface we will often use ['k] to refer to the capsule associated
     with that brand. *)
 
-(** Passwords represent access to a capsule. *)
+(** Converters can wrap and unwrap [Data.t] values from the current
+    capsule. *)
+module Converter : sig
+
+    (* CR layouts v5: this should have layout [void], but
+       [void] can't be used for function argument and return types yet. *)
+    type 'k t : value mod global portable many unique
+    (** ['k t] is the type of "converters" that can wrap and unwrap
+        [Data.t] values. An [uncontended] ['k t] indicates that ['k] is the
+        current capsule.  *)
+
+    type packed = P : 'k t -> packed
+    (** [packed] is the type of a converter for some unknown capsule.
+        Unpacking one provides a ['k t] together with a fresh
+        existential type brand for ['k]. *)
+
+end
+
+(** Obtain a converter for the current capsule. Since we do not
+    know the brand for the current capsule, we receive a fresh
+    one. *)
+val current : unit -> Converter.packed @@ portable
+
+(** The brand for the initial capsule. *)
+type initial
+
+(** A converter for the initial capsule *)
+val initial : initial Converter.t
+
+exception Contended of exn @@ contended
+(** If a function accessing the contents of the capsule raises an
+   exception, it is wrapped in [Contended] to avoid leaking access to
+   the data. *)
+
+(** Passwords represent permission to access to a capsule. *)
 module Password : sig
 
     (* CR layouts v5: this should have layout [void], but
        [void] can't be used for function argument and return types yet. *)
-    type 'k t
-    (** ['k t] is the type of "passwords" representing
-        the ability of the current domain to have [uncontended]
-        access to the capsule ['k]. They don't have a runtime representation,
-        and can't be passed between domains. Just as we don't share passwords
-        between people, we don't share [Password.t]s between domains.
-
-        Obtaining a ['k t] requires acquiring the mutex associated with ['k],
-        and modes prevent retaining the [t] after releasing the mutex. This
-        guarantees that uncontended access to the capsule is only granted
-        to a single domain at once. *)
+    type 'k t : value mod portable many unique uncontended
```

Comment:
Same for contention.

Reply:
Locality is not sufficient to guarantee non-shared access in presence of `fork_join`.

Reply:
Yeah, with `fork_join` we will be relying on the `unyielding/yielding` modes to prevent passwords from being used by two threads at the same time. You can think of a password as an effect handler, and `yielding` is how we prevent those from being used inside a `fork_join` too.